csharp/lang/correctness/double/double-epsilon-equality.yaml

```diff
@@ -0,0 +1,30 @@
+rules:
+- id: correctness-double-epsilon-equality
+  patterns: 
+    - pattern: |
+        $V1 - $V2
+    - pattern-inside: |
+        ... <= Double.Epsilon;
+    - pattern-not-inside: |
+        double $V1 = 0;
+        ...
+    - pattern-not-inside: |
+        double $V2 = 0;
+        ...
+    - pattern-not-inside: |
+        $V1 = 0;
+        ...
+    - pattern-not-inside: |
+        $V2 = 0;
+        ...
+  message: Double.Epsilon is defined by .NET as the smallest value that can be added to or subtracted from a zero-value Double.  It is unsuitable for equality comparisons of non-zero Double values.
```

Comment:
If I want to compare non-zero Double values, what should I use instead?

Reply:
You should use .NET's built-in equality rather than re-implementing it!  `Double.Epsilon`'s value is actually different across frameworks and systems (on ARM such as the M1, it's actually 0) - https://docs.microsoft.com/en-us/dotnet/api/system.double.epsilon?view=net-6.0#platform-notes

Custom double equality is "here there be dragons" behavior and should be avoided at all costs

Reply:
I assume this is the usual <, >, etc.? Can we add that as part of the message?

Reply:
@minusworld I think latest commit handles your comments, please verify

Reply:
Perfect 👍🏼